### PR TITLE
[Feature][exporter/azuredataexplorerexporter] Exporter helper support 

### PR DESCRIPTION
**Pull request by asaharn**
**Description:** <Describe what has changed.>

Added support for  Exporter Helper configuration. 
**Link to tracking Issue:** #24329 

**Testing:** <Describe what testing was performed and which tests were added.> Added tests and manually tested with e2e scenarios

### CODE REVIEW

**Comment by fatsheep9146** [comment]
changelog should be added

**Comment by fatsheep9146** [comment]
```
 ?   	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azuredataexplorerexporter/internal/metadata	[no test files]
--- FAIL: TestCreateMetricsExporter (0.01s)
    factory_test.go:52: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/azuredataexplorerexporter/factory_test.go:52
        	Error:      	An error is expected but got nil.
        	Test:       	TestCreateMetricsExporter
```

failed checks seemed related to this change, if so, please fix the test

**Comment by asaharn** [comment]
Hi @fatsheep9146 ,
Have updated the code with all the TCs passed. But looks like Lint check is failing with some other component.

File: exporter/azuredataexplorerexporter/factory.go [hunk 1]
```diff
@@ -75,6 +75,8 @@ func createMetricsExporter(
 		adxCfg,
 		adp.metricsDataPusher,
 		exporterhelper.WithTimeout(exporterhelper.TimeoutSettings{Timeout: 0}),
```
Comment (astencel-sumo) [inline exporter/azuredataexplorerexporter/factory.go:77]: Looks like you're still not using the `timeout` setting from `exporterhelper.TimeoutSettings`?

Comment (ag-ramachandran) [reply exporter/azuredataexplorerexporter/factory.go]: Ah! good catch @astencel-sumo, corrected this.

**Review by ag-ramachandran — APPROVED** [review]

File: cmd/githubgen/go.sum [hunk 2]
```diff
@@ -1,73 +0,0 @@
-github.com/ProtonMail/go-crypto v0.0.0-20230217124315-7d5c6f04bbb8 h1:wPbRQzjjwFc0ih8puEVAOFGELsn1zoIIYdxvML7mDxA=
```
Comment (fatsheep9146) [inline cmd/githubgen/go.sum:1]: This deleted file seems not related to this pr and cause the failed lint, could you revert this file?

Comment (asaharn) [reply cmd/githubgen/go.sum]: Reverted. Looks like this came in with mod tidy.

**Comment by asaharn** [comment]
Hi @fatsheep9146 , All checks passed now.

**Review by fatsheep9146 — APPROVED** [review]

**Comment by fatsheep9146** [comment]
@open-telemetry/collector-contrib-maintainer could anyone help merge this pr?

**Review by TylerHelmuth — APPROVED** [review]
